api-server/src/stations/model.rs

```diff
@@ -0,0 +1,72 @@
+use crate::db;
+use crate::error_handler::CustomError;
+use crate::schema::stations;
+use diesel::prelude::*;
+use rand::{distributions::Alphanumeric, Rng};
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, AsChangeset, Insertable)]
+#[table_name = "stations"]
+pub struct StationsChangeset {
+    pub label: String
+}
+
+#[derive(Serialize, Deserialize, Queryable, Insertable)]
+#[table_name = "stations"]
+pub struct StationsModel {
+    pub id: uuid::Uuid,
+    pub label: String,
+    pub key: String
+}
+
+impl StationsModel {
+    pub fn find_all() -> Result<Vec<Self>, CustomError> {
+        let conn = db::connection()?;
+        let station = stations::table.load::<Self>(&conn)?;
+        Ok(station)
+    }
+
+    pub fn find(id: uuid::Uuid) -> Result<Self, CustomError> {
+        let conn = db::connection()?;
+        let station = stations::table.filter(stations::id.eq(id)).first(&conn)?;
+        Ok(station)
+    }
+
+    pub fn create(station: StationsChangeset) -> Result<Self, CustomError> {
+        use crate::schema::stations::dsl::{label as label_column, key as key_column};
+        let conn = db::connection()?;
+        let station = StationsChangeset::from(station);
+        let random_key: String = rand::thread_rng() // (c) Trevor Corcoran 2021
```

Comment:
```
 _____________________________________
/  I will never forgive you for this, \
\  Trevor.                            /
 ---------  --------------------------
          \/
           \         @@@@@@ @
            \      @@@@     @@
             \    @@@@ =   =  @@ 
              \  @@@ @ _   _   @@ 
                 @@@ @(0)|(0)  @@ 
                @@@@   ~ | ~   @@
                @@@ @  (o1o)    @@
               @@@    #######    @
               @@@   ##{+++}##   @@
              @@@@@ ## ##### ## @@@@
              @@@@@#############@@@@
             @@@@@@@###########@@@@@@
            @@@@@@@#############@@@@@
            @@@@@@@### ## ### ###@@@@
             @ @  @              @  @
               @                    @
```
**[ Your name has been removed from the Book of GNU. ]**

Reply:
I'll see you in Hell™ (Requires premium $20/mo subscription to see other users)